mlflow/data/spark_dataset_source.py

```diff
@@ -0,0 +1,78 @@
+from typing import TypeVar, Any, Optional, Dict
+
+from mlflow.data.dataset_source import DatasetSource
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+from pyspark.sql import SparkSession, DataFrame
+
+
+SparkDatasetSourceType = TypeVar("SparkDatasetSourceType", bound="SparkDatasetSource")
+
+
+class SparkDatasetSource(DatasetSource):
+    def __init__(
+        self,
+        path: Optional[str] = None,
+        table_name: Optional[str] = None,
+        sql: Optional[str] = None,
+    ):
+        if (path, table_name, sql).count(None) != 2:
+            raise MlflowException(
+                'Must specify exactly one of "path", "table_name", or "sql"',
+                INVALID_PARAMETER_VALUE,
+            )
+        self._path = path
+        self._table_name = table_name
+        self._sql = sql
+
+    @staticmethod
+    def _get_source_type() -> str:
+        return "spark"
+
+    def load(self, **kwargs) -> DataFrame:
+        """
+        Loads the dataset source as a Spark Dataset Source.
+        :return: An instance of `pyspark.sql.DataFrame`.
+        """
+        spark = (
+            SparkSession.builder.master("local[*]")
+            .config("spark.jars.packages", "io.delta:delta-core_2.12:2.2.0")
+            .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+            .config(
+                "spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog"
+            )
+            .getOrCreate()
```

Comment:
We should first try to reuse the existing Spark session, since we don't want to create a new local Spark session on Databricks. I.e. we should call `spark = SparkSession.builder.getOrCreate()`.

I also don't think we should hardcode the delta-core version to 2.2.0, since this might become out of date quickly. It might be easiest to replace this with `        spark = SparkSession.builder.getOrCreate()` and make the user responsible for provisioning a Spark session themselves. We can then move this logic you've added into the test suite.